.github/workflows/validate_dataset.yml

```diff
@@ -0,0 +1,68 @@
+name: Validate Datasets
+
+# on .json PR to /data/datasets folder
+on: 
+  push:
+    secrets:
+      stac_ingestor_url:
+        required: true
+      client_id:
+        required: true
+      client_secret:
+        required: true
+      cognito_domain:
+        required: true
+      scope:
+        required: true
+    paths:
+      - 'data/datasets/**'
+
+# send pushed file as body to (API)/datasets/validate using secret credentials
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Filter for dataset files
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          base: 'main'
+          list-files: shell
+          filters: |
+            dataset:
+              - 'data/datasets/*.json'
```

Comment:
probably just get the added/updated files in `data/datasets/` folder?
using something like [this](https://github.com/marketplace/actions/get-all-changed-files) and a filter command.